packages/material-ui/src/Stepper/Stepper.js

```diff
@@ -83,10 +83,9 @@ function Stepper(props) {
         index > 0 &&
         React.cloneElement(connector, {
           key: index, // eslint-disable-line react/no-array-index-key
-          index,
           ...state,
         }),
-      React.cloneElement(step, { ...controlProps, ...step.props, ...state }),
+      React.cloneElement(step, { ...controlProps, ...state, ...step.props }),
```

Comment:
Allow people to override the value.

Reply:
The behavior wrongly changed in https://github.com/mui-org/material-ui/pull/13023/files#diff-17aef418c2c1e2df041cebf7c67ce4cbR88.